src/Controls/tests/DeviceTests/Elements/CollectionView/CollectionViewTests.cs

```diff
@@ -138,6 +139,91 @@ public async Task ItemsSourceDoesNotLeak()
 			Assert.True(logicalChildren.Count <= 5, "_logicalChildren should not grow in size!");
 		}
 
+		[Fact]
+		public async Task CollectionScrollToGroupWorks()
+		{
+			SetupBuilder();
+
+			var dataList = new List<TestData>();
+			string letters = "nice"; // 4 groups
+			for (int i = 0; i < letters.Length; i++)
+			{
+				for (int n = 0; n < 10; n++)
+				{
+					dataList.Add(new TestData
+					{
+						Name = $"{letters[i]}_{n + 1}"
+					});
+				}
+			}
+
+			var grouped =
+				  from p in dataList
+				  orderby p.Name
+				  group p by p.Name[0].ToString()
+				  into groups
+				  select
+					   new TestDataGroup(groups.Key, groups.ToList());
+
+			var collectionView = new CollectionView
+			{
+				IsGrouped = true,
+				ItemsSource = grouped.ToList(),
+				ItemTemplate = new DataTemplate(() => 
+				{
+					var name = new Label()
+					{
+						TextColor = Colors.Grey
+					};
+					name.SetBinding(Label.TextProperty, "Name");
+					return name;
+				}),
+				GroupHeaderTemplate = new DataTemplate(() =>
+				{
+					var name = new Label()
+					{
+						TextColor = Colors.White,
+						FontSize = 18,
+						FontAttributes = FontAttributes.Bold
+					};
+					name.SetBinding(Label.TextProperty, "Name");
+					return name;
+				})
+			};
+
+			var lastVisibleItemIndex = -1;
+			collectionView.Scrolled += (s, e) =>
+			{
+				lastVisibleItemIndex = e.LastVisibleItemIndex;
+			};
+
+			await CreateHandlerAndAddToWindow<CollectionViewHandler>(collectionView, async handler =>
+			{
+				collectionView.ScrollTo(index: 4, groupIndex: 3); // Group 'c'
+
+				await Task.Delay(100);
+
+				// each group has 10 items, plus the 3 headers
+				Assert.True(lastVisibleItemIndex == 34);
```

Comment:
The test is failing on Android API_33:
![image](https://github.com/dotnet/maui/assets/6755973/fb5f73cf-4359-4021-a99c-37261e94756e)